From: Debian 3-D Printing Packages
 <3dprinter-general@lists.alioth.debian.org>
Date: Tue, 17 May 2022 12:27:42 -0600
Subject: relax-test

Allow some precision deviation on i386

[ RUN      ] ArcusCommunicationPrivateTest.ReadMeshGroupMessage
/<<PKGBUILDDIR>>/tests/arcus/ArcusCommunicationPrivateTest.cpp:250: Failure
Expected equality of these values:
  max_coords[i] - min_coords[i]
    Which is: 9900
  raw_max_coords[i] - raw_min_coords[i]
    Which is: 9898
/<<PKGBUILDDIR>>/tests/arcus/ArcusCommunicationPrivateTest.cpp:250: Failure
Expected equality of these values:
  max_coords[i] - min_coords[i]
    Which is: 9900
  raw_max_coords[i] - raw_min_coords[i]
    Which is: 9898
/<<PKGBUILDDIR>>/tests/arcus/ArcusCommunicationPrivateTest.cpp:250: Failure
Expected equality of these values:
  max_coords[i] - min_coords[i]
    Which is: 9900
  raw_max_coords[i] - raw_min_coords[i]
    Which is: 9899
[  FAILED  ] ArcusCommunicationPrivateTest.ReadMeshGroupMessage (1 ms)
[----------] 4 tests from ArcusCommunicationPrivateTest (2 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 1 test suite ran. (2 ms total)
[  PASSED  ] 3 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ArcusCommunicationPrivateTest.ReadMeshGroupMessage

 1 FAILED TEST

https://github.com/Ultimaker/CuraEngine/issues/1192
---
 tests/arcus/ArcusCommunicationPrivateTest.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/arcus/ArcusCommunicationPrivateTest.cpp b/tests/arcus/ArcusCommunicationPrivateTest.cpp
index 8c958ed..5010909 100644
--- a/tests/arcus/ArcusCommunicationPrivateTest.cpp
+++ b/tests/arcus/ArcusCommunicationPrivateTest.cpp
@@ -247,7 +247,11 @@ TEST_F(ArcusCommunicationPrivateTest, ReadMeshGroupMessage)
     // - Then, just compare:
     for (int i = 0; i < 3; ++i)
     {
+#ifdef __i386__
+        EXPECT_LE(abs((max_coords[i] - min_coords[i]) - (raw_max_coords[i] - raw_min_coords[i])), 2);
+#else
         EXPECT_EQ(max_coords[i] - min_coords[i], raw_max_coords[i] - raw_min_coords[i]);
+#endif
     }
 }
 
